### PR TITLE
NEPT-1133: Integrate the ECL component into Drupal form textfield and password

### PR DESCRIPTION
**Pull request by gillesdeudon**

### CODE REVIEW

File: templates/password/password.component.inc [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * @file
+ * Contains password component file.
+ */
+
+/**
+ * Implements hook_preprocess_HOOK() for the form password.
+ */
+function europa_preprocess_password(&$variables, $hook) {
+  $element = $variables['element'];
```
Comment (voidtek) [inline templates/password/password.component.inc]: Should be "$element = &$variables['element'];"

File: templates/textfield/textfield.component.inc [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * @file
+ * Contains texfield component file.
+ */
+
+/**
+ * Implements hook_preprocess_HOOK() for the form textfield.
+ */
+function europa_preprocess_textfield(&$variables, $hook) {
+  $element = $variables['element'];
```
Comment (voidtek) [inline templates/textfield/textfield.component.inc]: Should be "$element = &$variables['element'];"

File: templates/password/password.component.inc (hunk 1 above)
Comment (gillesdeudon) [inline templates/password/password.component.inc]: No really needed here as it is never altered.

File: templates/textfield/textfield.component.inc (hunk 2 above)
Comment (gillesdeudon) [inline templates/textfield/textfield.component.inc]: No really needed here as it is never altered.

**Comment by voidtek** [comment]
:+1:

File: template.php [hunk 3]
```diff
@@ -76,3 +76,29 @@ function _europa_term_heading($element) {
...
+  if (!empty($classes)) {
+    if (!isset($element['#attributes']['class'])) {
+      $element['#attributes']['class'] = array();
+    }
+    $element['#attributes']['class'] = array_merge($element['#attributes']['class'], $classes);
+  }
+
+  // Determines if the error class must added.
+  // The logic comes from the Drupal function: "_form_set_class".
+  if (isset($element['#parents']) && form_get_error($element) !== NULL && !empty($element['#validated'])) {
```
Comment (netlooker) [inline template.php]: This part requires further modifications when the complete input components list become available.

**Comment by netlooker** [comment]
👍

**Pull request merged**